### ISSUE TITLE
`cargo-spellcheck` silently skips files with leading HTML blocks

### ISSUE THREAD

**Issue by LesnyRumcajs** [drahnr/cargo-spellcheck#357]
**Describe the bug**

When a Markdown file starts with an HTML block (e.g. `<p>`, `<div>`), `cargo-spellcheck` silently skips all subsequent content in the file. No spelling errors are reported, even for obviously misspelled words.

**To Reproduce**

1. Create a file `with_html.md`:

```md
<p align="center">
  <img src="logo.png">
</p>

This contains a xyzzyspam misspelling.
```

2. Create a minimal config `spellcheck.toml`:

```toml
[hunspell]
skip_os_lookups = true
use_builtin = true

[hunspell.quirks]
transform_regex = []
```

3. Run:

```
$ cargo spellcheck check --code 1 -c spellcheck.toml with_html.md
```

4. Observe: no errors reported, exit code 0.

5. Now remove the leading HTML block:

```md
This contains a xyzzyspam misspelling.
```

6. Run the same command — `xyzzyspam` is correctly flagged.

**Expected behavior**

Both files should report `xyzzyspam` as a spelling mistake. Leading HTML blocks should not cause the rest of the file's content to be silently skipped.

**Screenshots**

```
$ cargo spellcheck check --code 1 -c spellcheck.toml with_html.md
$ echo $?
0

$ cargo spellcheck check --code 1 -c spellcheck.toml without_html.md
error: spellcheck(Hunspell)
  --> without_html.md:1
   |
 1 | This contains a xyzzyspam misspelling.
   |                 ^^^^^^^^^
   |   Possible spelling mistake found.
```

**Please complete the following information:**
 - System: Fedora 42
 - Obtained: cargo (via `cargo binstall`)
 - Version: `cargo-spellcheck 0.15.5`

**Additional context**

This is a common pattern in `README.md` files that use centered HTML badges at the top (e.g. GitHub project badges). It effectively makes spellchecking the entire file a no-op.

Verbose trace (`-vvvv`) confirms the text *is* extracted by the CommonMark parser but never sent to the Hunspell checker:

```

TRACE doc_chunks::markdown] Parsing event: Start(HtmlBlock)
TRACE doc_chunks::markdown] Parsing event: Html(Borrowed("<p align=\"center\">\n"))
...
TRACE doc_chunks::markdown] Parsing event: End(HtmlBlock)
TRACE doc_chunks::markdown] Parsing event: Text(Borrowed("This contains a xyzzyspam misspelling."))
```

The text is parsed but no corresponding `Checking xyzzyspam` trace appears.

Single repro script:
```shell
#!/usr/bin/env bash
set -euo pipefail

# Minimal reproduction for cargo-spellcheck silently skipping content
# after leading HTML blocks in markdown files.

cd "$(dirname "$0")"

# Minimal config — no custom dictionaries, no regex transforms
cat > spellcheck.toml << 'TOML'
[hunspell]
skip_os_lookups = true
use_builtin = true

[hunspell.quirks]
transform_regex = []
TOML

# Markdown file WITH leading HTML block
cat > with_html.md << 'MD'
<p align="center">
  <img src="logo.png">
</p>

This contains a xyzzyspam misspelling.
MD

# Identical content WITHOUT leading HTML block
cat > without_html.md << 'MD'
This contains a xyzzyspam misspelling.
MD

echo "=== WITH leading HTML block ==="
cargo spellcheck check --code 1 -c spellcheck.toml with_html.md 2>&1 || true
echo

echo "=== WITHOUT leading HTML block ==="
cargo spellcheck check --code 1 -c spellcheck.toml without_html.md 2>&1 || true
echo

echo "=== Expected: both should report 'xyzzyspam' as misspelled ==="
```
